### PR TITLE
upgrade build tools and enable r8 full mode

### PR DESCRIPTION
**Pull request by connyduck**
~250kB smaller build

### CODE REVIEW

File: app/src/main/res/layout/fragment_view_video.xml [hunk 1]
```diff
@@ -12,7 +12,6 @@
         android:id="@+id/mediaDescription"
         android:layout_width="match_parent"
         android:layout_height="wrap_content"
-        app:layout_constraintTop_toBottomOf="@+id/toolbar"
```
Comment (connyduck) [inline app/src/main/res/layout/fragment_view_video.xml]: lint was blocking release builds because of this warning

Comment (charlag) [reply app/src/main/res/layout/fragment_view_video.xml]: Is it okay? Will it overlap?

**Comment by connyduck** [comment]
As of Build Tools 3.4, R8 is the default shrinker, nothing else to do. `android.enableR8.fullMode=true` enables even more optimizations, but removes compatibility with ProGuard.

File: app/src/main/res/layout/fragment_view_video.xml (hunk 1 above)
Comment (connyduck) [reply app/src/main/res/layout/fragment_view_video.xml]: Oh, looks like this was never actually working. I'll fix it.

**Comment by charlag** [comment]
Do we need to add some rules for it then? We use Retrofit so we have some dynamicity.

**Comment by connyduck** [comment]
Yes, R8 uses ProGuard rules, we leave them in the project. Or do you mean something else?

**Comment by charlag** [comment]
Nah, I was just concerned about this moment. Good go go then I guess!

**Pull request merged**